server/src/main/java/com/thoughtworks/go/server/service/MaterialService.java

```diff
@@ -160,4 +160,16 @@ private void resolveSecretParams(Material material) {
     Class<? extends Material> getMaterialClass(Material material) {
         return material.getClass();
     }
+
+    public Map<String, Modification> getModificationWithMaterial() {
+        List<Modification> modifications = materialRepository.getModificationWithMaterial();
+        Map<String, Modification> materialAndModifications = new HashMap<>();
+        modifications.forEach((mod) -> {
+            String fingerprint = mod.getMaterialInstance().getFingerprint();
+            Modification modification = new Modification(mod);
+            modification.setMaterialInstance(null);
```

Comment:
Why do we need to create a new Modification object and set a null MaterialInstance?

Reply:
There was no use of the material instance information so I set it to null.

Reply:
I would consider using streams and just convert the list to hash, rather than building a new modification object.